core/auth/domain/auth.go

```diff
@@ -9,32 +12,67 @@ type (
 	}
 
 	// User is a basic authenticated user
-	User struct {
-		Sub          string
-		Name         string
-		Email        string
-		Salutation   string
-		FirstName    string
-		LastName     string
-		Street       string
-		ZipCode      string
-		City         string
-		DateOfBirth  string
-		Country      string
-		CustomFields map[string]string
-		Groups       []string
+	User interface {
+		Subject() string
+		Email() (string,bool)
```

Comment:
I would go with single return argument, with default value as empty string. In general it would be easier for further usage and easy to check for empty string if it's needed.